### PR TITLE
Split out style from LocationSupportableMapView

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [X] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Removed style from LocationSupportableMapView</changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This removes `LocationSupportableMapView.style`, moving it to a new `LocationStyleDelegate` (naming mirroring `AnnotationStyleDelegate`.

This is a prelude to being able to move `Style` from `MapView` to `MapboxMap`.

### CODE REVIEW

**Comment by nishant-karajgikar** [comment]
Hmm it feels like LocationSupportableMapView is basically obsolete?

This is what locationSupportableMapView currently has:

```swift
public protocol LocationSupportableMapView: UIView {

    /// Matching the style property in `MapView`
    var style: Style! { get }

    /// Returns the screen coordinate for a given location coordinate (lat-long)
    func screenCoordinate(for locationCoordinate: CLLocationCoordinate2D) -> ScreenCoordinate

    /// Allows the `LocationSupportableMapView` to subscribe to a delegate
    func subscribeRenderFrameHandler(_ handler: @escaping (MapboxCoreMaps.Event) -> Void)

    /// Allows the `LocationSupportableMapView` to subscrive to a delegate function and handle style change events
    func subscribeStyleChangeHandler(_ handler: @escaping (MapboxCoreMaps.Event) -> Void)

    /// Gets meters per point at latitude for calculating accuracy ring
    func metersPerPointAtLatitude(latitude: CLLocationDegrees) -> CLLocationDistance

}

```

All these properties will soon be available on `mapboxMap` (if not alread -- like the event api)..

**Comment by julianrex** [comment]
@nishant-karajgikar Yes, I think it will soon disappear, or mutate.

File: Tests/MapboxMapsTests/Location/MockLocationStyleDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    func removeLayer(withId id: String) throws {}
+    func layerExists(withId id: String) -> Bool { return false }
+    func setLayerProperties(for layerId: String, properties: [String: Any]) throws {}
+    func addSource(_ source: Source, id: String) throws {}
+    func removeSource(withId id: String) throws {}
+    func setSourceProperty(for sourceId: String, property: String, value: Any) throws {}
+
+    //swiftlint:disable function_parameter_count
+    func addImage(_ image: UIImage, id: String, sdf: Bool, stretchX: [ImageStretches], stretchY: [ImageStretches], content: ImageContent?) throws {}
+}
```
Comment (nishant-karajgikar) [inline Tests/MapboxMapsTests/Location/MockLocationStyleDelegate.swift:15]: Could we add stubs (`Stub<>`) here? Or add a ticket to add stubs as part of the location re-design that @neelmistry94  is doing?

**Review by nishant-karajgikar — APPROVED** [review]

File: Tests/MapboxMapsTests/Location/MockLocationStyleDelegate.swift (hunk 1 above)
Comment (neelmistry94) [reply Tests/MapboxMapsTests/Location/MockLocationStyleDelegate.swift:15]: I will add it as part of my work

**Review by neelmistry94 — APPROVED** [review]

File: Tests/MapboxMapsTests/Location/MockLocationStyleDelegate.swift (hunk 1 above)
Comment (julianrex) [reply Tests/MapboxMapsTests/Location/MockLocationStyleDelegate.swift:15]: thank you!

**Pull request merged**